src/main/java/net/snowflake/ingest/streaming/internal/DataValidationUtil.java

```diff
@@ -802,6 +802,83 @@ static double validateAndParseReal(String columnName, Object input, long insertR
         columnName, input.getClass(), "REAL", new String[] {"Number", "String"}, insertRowIndex);
   }
 
+  /**
+   * Validates and parses input to a specific Iceberg column number type including INT, LONG, FLOAT,
+   * and DOUBLE. Allowed Java types:
+   *
+   * <ul>
+   *   <li>Number
+   *   <li>String
+   * </ul>
+   *
+   * @param columnName Column name, used in validation error messages
+   * @param input Object to validate and parse
+   * @param insertRowIndex Row index for error reporting
+   * @param targetType The type to which the input should be parsed ("INT", "LONG", "FLOAT",
+   *     "DOUBLE")
+   * @return Parsed number as Number to be casted by the caller
+   */
+  static <T extends Number> T validateAndParseIcebergNumber(
+      String columnName, Object input, long insertRowIndex, Class<T> targetType) {
+    if (input instanceof Number) {
+      if (targetType.equals(Integer.class)) {
+        return targetType.cast(((Number) input).intValue());
+      } else if (targetType.equals(Long.class)) {
+        return targetType.cast(((Number) input).longValue());
+      } else if (targetType.equals(Float.class)) {
+        return targetType.cast(((Number) input).floatValue());
+      } else if (targetType.equals(Double.class)) {
+        return targetType.cast(((Number) input).doubleValue());
+      }
+    } else if (input instanceof String) {
+      String stringInput = ((String) input).trim().toLowerCase();
```

Comment:
why toLowerCase?

Reply:
only do this when truly necessary (in the catch block I believe)

Reply:
Removed this function and use `validateAndParseReal` for float and double parsing.